File: src/main/java/de/digitalstep/ntlmproxy/ui/ApplicationProperties.java
package de.digitalstep.ntlmproxy.ui;

import java.io.IOException;
import java.util.Properties;

public class ApplicationProperties {

    private static final ApplicationProperties INSTANCE = new ApplicationProperties();

    public static ApplicationProperties applicationProperties() {
        return INSTANCE;
    }

    private final Properties properties = new Properties();

    private ApplicationProperties() {
        try {
            properties.load(getClass().getResourceAsStream("/application.properties"));
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    public String getApplicationName() {
        return get("application.name");
    }

    public String getApplicationVersion() {
        return get("application.version");
    }

    private String get(String property) {
        return properties.getProperty(property);
    }

}

File: src/main/scala/de/digitalstep/ntlmproxy/ui/ApplicationProperties.scala
package de.digitalstep.ntlmproxy.ui

import java.util.Properties

object ApplicationProperties {
	private val properties = new Properties
	properties.load(getClass().getResourceAsStream("/application.properties"))

	val applicationName = properties.getProperty("application.name")
	val applicationVersion = properties.getProperty("application.version")

}